gnd/src/main/java/com/google/android/gnd/ui/offlinearea/selector/OfflineAreaSelectorFragment.java

```diff
@@ -55,6 +60,21 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     // TODO: use the viewmodel
     Single<MapAdapter> mapAdapter = mapProvider.getMapAdapter();
     mapAdapter.as(autoDisposable(this)).subscribe(this::onMapReady);
+    viewModel.getDownloadMessages().observe(this, e -> e.ifUnhandled(this::onDownloadMessage));
+  }
+
+  private void onDownloadMessage(DownloadMessage message) {
+    switch (message) {
+      case STARTED:
+        EphemeralPopups.showSuccess(getContext(), R.string.offline_area_download_started);
+        navigator.navigateUp();
+        break;
+      case FAILURE:
+      default:
```

Comment:
Why default? Doesn't this make above line `case FAILURE` redundant?

Reply:
It does, but I believe PMD (or some other check) requires a default clause. We could drop the `FAILURE` clause, but I think it's ok to keep for now in case we decide we want to handle that case differently--this way we're explicit about discriminating over all the enum values.